server/src/main/java/org/jboss/as/server/jmx/MBeanServerPlugin.java

```diff
@@ -51,4 +53,14 @@
      * @return whether or not this plugin should audit log
      */
     boolean shouldAuthorize();
+
+    /**
+     * Return the operation target for calling this method on this object.
+     * @param name
+     * @param method
+     * @return
+     */
+    JmxTarget getJmxTarget(final ObjectName name, final String method);
+
+    JmxAction getJmxAction(final ObjectName name, final String method, String operation, JmxAction.Impact impact, boolean isNonFacadeMBeansSensitive, String attribute);
```

Comment:
The javadoc should include this

* @throws IllegalStateException if {@link #shouldAuthorize()} returns {@code false}

Also, please fill in the rest of the javadoc.